### PR TITLE
Complete homework 4 for chapter 1.

### PR DESCRIPTION
**Pull request by kungfuphil**
Completed Chapter 1, homework 4.

I did change one of the lines in the Lesson3Tests.cs file, though, because the wording was inconsistent between the first test and last one.

[InlineData(100, 0, "Height cannot be **equal or** less than zero, but was 0.")]
[InlineData(0, 50, "Weight cannot be equal or less than zero, but was 0.")]
[InlineData(-100, 50, "Weight cannot be equal or less than zero, but was -100.")]
~~[InlineData(0, 0, "Weight cannot be equal or less than zero, but was 0.Height cannot be less than zero, but was 0.")]~~
[InlineData(0, 0, "Weight cannot be equal or less than zero, but was 0.Height cannot be **equal or** less than zero, but was 0.")]

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            GetInfoAndPrint();
+        }
+
+        static void GetInfoAndPrint()
+        {
+            Console.Write("Name: ");
+            string name = Console.ReadLine();
+
+            var age = PromptForInt("Age: ");
+
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs:20]: No need to have a blank line after related pieces of code. I would separate calculation and printing, but the rest seems related.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            var invC = CultureInfo.InvariantCulture;
+
+            Console.WriteLine(message);
+            string input = Console.ReadLine();
+            bool isNumber = float.TryParse(input, NumberStyles.Float, invC, out float number);
+            
+            if (!isNumber && !string.IsNullOrEmpty(input))
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs:78]: -1 is a magic arbitary number. Please use a constant for that to explain its' meaning to me and other readers.

**Review by Almantask — APPROVED** [review]
Overall looks good!

**Comment by Almantask** [comment]
> Completed Chapter 1, homework 4.
> 
> I did change one of the lines in the Lesson3Tests.cs file, though, because the wording was inconsistent between the first test and last one.
> 
> [InlineData(100, 0, "Height cannot be **equal or** less than zero, but was 0.")]
> [InlineData(0, 50, "Weight cannot be equal or less than zero, but was 0.")]
> [InlineData(-100, 50, "Weight cannot be equal or less than zero, but was -100.")]
> ~[InlineData(0, 0, "Weight cannot be equal or less than zero, but was 0.Height cannot be less than zero, but was 0.")]~
> [InlineData(0, 0, "Weight cannot be equal or less than zero, but was 0.Height cannot be **equal or** less than zero, but was 0.")]

Yeah, it's a good choice and I am aware of inconsistencies- it annoy me as well. Never have the time to actually go and make the changes myself 😢 

Well done for spotting it!

**Comment by kungfuphil** [comment]
Thank you for the comments! I can submit a pull request with just the update to the InlineData where the comment is different, if that works for you.

**Pull request closed without merging**